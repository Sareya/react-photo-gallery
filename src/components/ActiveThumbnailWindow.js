/** @jsx jsx */
import React from 'react'
import { jsx } from '@emotion/core'


function renderTextContent(thumbnail) {


let colorIndex = thumbnail.url.split('/')
let color = colorIndex[4]
let colorHex = '#' + color
console.log(colorHex);

return(
  <React.Fragment>
    <h2 style={{paddingLeft:'20px',color:colorHex}}>Image Id</h2>
    <p style={{paddingLeft:'20px',fontSize:'30px'}}>{thumbnail.id}</p>
    <h2 style={{paddingLeft:'20px',color:colorHex}}>Album Id</h2>
    <p style={{paddingLeft:'20px',fontSize:'30px'}}>{thumbnail.albumId}</p>
    <h2 style={{paddingLeft:'20px',color:colorHex}}>Image Description </h2>
    <p style={{paddingLeft:'20px',fontSize:'30px'}}>{thumbnail.title}</p>

  </React.Fragment>

)

}

const ActiveThumbnailWindow = ( props ) => {

return(
  <div style={{background: '#ddd',
  width: '1024px',
  margin: '40px auto',
  display: 'flex'}}>
      <div style={{flex:'1'}}><img style ={{height:'100%'}} src={props.activeThumbnail.url} alt='Nothing to show'/>
      </div>
      <div style={{flex:'1'}}>{renderTextContent(props.activeThumbnail)}</div>
    </div>
  )


}

export default ActiveThumbnailWindow
