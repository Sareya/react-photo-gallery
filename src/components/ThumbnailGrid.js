/** @jsx jsx */
import {css, jsx } from '@emotion/core'

export const ThumbnailGrid = ({ thumbnails, onClick }) => (
  <div
    css={css`
      height: 45%;
      width: 100%;
      display: flex;
      flex-wrap: wrap;
    `}
  >
    {thumbnails.map((thumbnail, i) => (
      <Thumbnail
        key={thumbnail.thumbnailUrl}
        imgUrl={thumbnail.thumbnailUrl}
        index={i}
        onClick={onClick}
      />
    ))}
  </div>
)

const Thumbnail = ({ imgUrl, index, onClick }) => (
  <div
    css={css`
      height: 50%;
      width: 25%;
      position: relative;
      cursor: pointer;
      img {
        width: 100%;
        height: 100%;
      }
    `}
  >
    <img src={imgUrl} alt= 'Nothing to display' data-index={index} onClick={onClick} />
  </div>
)
