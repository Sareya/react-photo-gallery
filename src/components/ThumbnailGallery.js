/** @jsx jsx */
import React, { useState , useEffect} from 'react'
import axios from 'axios'
import { jsx } from '@emotion/core'
import InfiniteScroll from 'react-infinite-scroll-component';
import { Link} from 'react-router-dom'
import { ThumbnailGrid } from './ThumbnailGrid'


const ThumbnailGallery = (props) => {

  const [thumbnails, setThumbnails] = useState([])
  const [activeIndex, setActiveIndex] = useState(0)
  let [page, setPage] = useState(1)

  useEffect((page) => {
    getPhotos(page)
  }, [])

  const getPhotos = (p) => {
    console.log('getPhotos ' + page)

    axios
      .get(
        `https://jsonplaceholder.typicode.com/photos?_page=${page}&_limit=10`
      )
      .then(res => {
        setThumbnails(thumbnails => [...thumbnails, ...res.data]);
      });


  }

  function RenderThumbnails(){
    return (
      thumbnails.length ? (
           <InfiniteScroll style={{margin:'60px'}}
            dataLength={thumbnails.length}
            next={FetchMoreData}
            hasMore={true}
            loader={<h4>Loading...</h4>}>
              <Link to="/second">
              <ThumbnailGrid thumbnails={thumbnails} onClick={handleClick} />
              </Link>
           </InfiniteScroll>

): null
    ) }


  const FetchMoreData = () => {

      page=page+1
      console.log(page)
      setPage(page)
      console.log(page)
      getPhotos(page)


  };

  function handleClick (e){

    const activeIndex = e.target.getAttribute('data-index')
    setActiveIndex(activeIndex)
    props.setActiveThumbnail(thumbnails[activeIndex])
  }

return(

    <React.Fragment>

          <RenderThumbnails/>

    </React.Fragment>
)}

export default ThumbnailGallery
