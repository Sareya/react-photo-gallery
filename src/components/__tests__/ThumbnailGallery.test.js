import ThumbnailGallery from '../ThumbnailGallery'
import ThumbnailGrid from '../ThumbnailGrid'
import ActiveThumbnailWindow from '../ActiveThumbnailWindow'
import renderer from 'react-test-renderer';
import React, { useState } from 'react';
import axios from 'axios';
import { mount, shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
// import configureStore from 'redux-mock-store';
// import sinon from 'sinon';
import { StaticRouter } from 'react-router-dom'
import expect from "expect";


jest.mock('axios')
configure({ adapter: new Adapter() });
// const mockStore = configureStore();
// sinon.spy(ThumbnailGallery.prototype, 'useEffect');

describe('Snapshot Testing', () => {

  it('Renders ThumbnailGallery snapshot as expected', () => {
          const tree = renderer.create(<ThumbnailGallery />).toJSON();
          expect(tree).toMatchSnapshot();
  });

});


// test('Renders ActiveThumbnailWindow snapshot as expected', () => {
//   const thumbnails={
//                       "albumId": 1,
//                       "id": 1,
//                       "title": "accusamus beatae ad facilis cum similique qui sunt",
//                       "url": "https://via.placeholder.com/600/92c952",
//                       "thumbnailUrl": "https://via.placeholder.com/150/92c952"
//                         }
//
//         const tree = renderer.create(<ActiveThumbnailWindow activeThumbnail={thumbnails}/>).toJSON();
//         expect(tree).toMatchSnapshot();
// });

describe('getPhotos', () => {
  it('fetches successfully data from an API', async () => {
    const data = [{
                        "albumId": 1,
                        "id": 1,
                        "title": "accusamus beatae ad facilis cum similique qui sunt",
                        "url": "https://via.placeholder.com/600/92c952",
                        "thumbnailUrl": "https://via.placeholder.com/150/92c952"
                      },{
                        "albumId": 1,
                        "id": 2,
                        "title": "reprehenderit est deserunt velit ipsam",
                        "url": "https://via.placeholder.com/600/771796",
                        "thumbnailUrl": "https://via.placeholder.com/150/771796"
                      },{
                        "albumId": 1,
                        "id": 3,
                        "title": "officia porro iure quia iusto qui ipsa ut modi",
                        "url": "https://via.placeholder.com/600/24f355",
                        "thumbnailUrl": "https://via.placeholder.com/150/24f355"
                      }
]

    axios.get.mockImplementationOnce(() => Promise.resolve(data));

  });

  it('fetches erroneously data from an API', async () => {
    const errorMessage = 'Network Error';

    axios.get.mockImplementationOnce(() =>
      Promise.reject(new Error(errorMessage)),
    );
  });
});
