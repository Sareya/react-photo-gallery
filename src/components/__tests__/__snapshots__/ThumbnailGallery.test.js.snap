// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ActiveThumbnailWindow snapshot as expected 1`] = `
<div
  style={
    Object {
      "background": "#ddd",
      "display": "flex",
      "margin": "40px auto",
      "width": "1024px",
    }
  }
>
  <div
    style={
      Object {
        "flex": "1",
      }
    }
  >
    <img
      alt="Nothing to show"
      src="https://via.placeholder.com/600/92c952"
      style={
        Object {
          "height": "100%",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "flex": "1",
      }
    }
  >
    <h2
      style={
        Object {
          "color": "#92c952",
          "paddingLeft": "20px",
        }
      }
    >
      Image Id
    </h2>
    <p
      style={
        Object {
          "fontSize": "30px",
          "paddingLeft": "20px",
        }
      }
    >
      1
    </p>
    <h2
      style={
        Object {
          "color": "#92c952",
          "paddingLeft": "20px",
        }
      }
    >
      Album Id
    </h2>
    <p
      style={
        Object {
          "fontSize": "30px",
          "paddingLeft": "20px",
        }
      }
    >
      1
    </p>
    <h2
      style={
        Object {
          "color": "#92c952",
          "paddingLeft": "20px",
        }
      }
    >
      Image Description 
    </h2>
    <p
      style={
        Object {
          "fontSize": "30px",
          "paddingLeft": "20px",
        }
      }
    >
      accusamus beatae ad facilis cum similique qui sunt
    </p>
  </div>
</div>
`;

exports[`Renders ThumbnailGallery snapshot as expected 1`] = `null`;

exports[`Renders snapshot as expected 1`] = `null`;

exports[`Snapshot Testing Renders ThumbnailGallery snapshot as expected 1`] = `null`;
